toolchain/check/testdata/impl/lookup/min_prelude/struct.carbon

```diff
@@ -0,0 +1,846 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/min_prelude/facet_types.carbon
+// EXTRA-ARGS: --custom-core
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/impl/lookup/min_prelude/struct.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/lookup/min_prelude/struct.carbon
+
+interface Z {
+  let X:! type;
+  fn ZZ() -> X;
+}
+
+class C { adapt (); }
+class D { adapt (); }
+
+impl {.a: (), .b: ()} as Z where .X = C {
+  fn ZZ() -> C { return () as C; }
+}
+// Structs with different field names are different types.
+impl {.aa: (), .bb: ()} as Z where .X = D {
+  fn ZZ() -> D { return () as D; }
+}
+
+// Are structs with different field orders different types too, for impl lookup?
```

Comment:
Could this comment start with "TODO" or "Open question:" or something like that?

Reply:
Done